presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/BridgingHiveMetastore.java

```diff
@@ -319,15 +321,22 @@ private void alterTable(HiveIdentity identity, String databaseName, String table
     }
 
     @Override
-    public Optional<List<String>> getPartitionNames(HiveIdentity identity, String databaseName, String tableName)
+    public Optional<List<String>> getPartitionNamesByFilter(
+            HiveIdentity identity,
+            String databaseName,
+            String tableName,
+            List<String> columnNames,
+            TupleDomain<String> partitionKeysFilter)
     {
-        return delegate.getPartitionNames(identity, databaseName, tableName);
-    }
+        if (partitionKeysFilter.isNone()) {
+            return Optional.of(ImmutableList.of());
+        }
 
-    @Override
-    public Optional<List<String>> getPartitionNamesByParts(HiveIdentity identity, String databaseName, String tableName, List<String> parts)
-    {
-        return delegate.getPartitionNamesByParts(identity, databaseName, tableName, parts);
+        if (partitionKeysFilter.isAll()) {
+            return delegate.getPartitionNames(identity, databaseName, tableName);
```

Comment:
Should we drop the `getPartitionNames()` method from the `ThriftMetastore` interface? Then this can be a straight delegation.